__tests__/__unit__/dataset/actions.unit.test.ts

```diff
@@ -0,0 +1,1311 @@
+};
+
+Object.defineProperty(vscode.window, "withProgress", { value: jest.fn() });
+Object.defineProperty(zowe, "Upload", { value: jest.fn() });
+Object.defineProperty(zowe.Upload, "bufferToDataSet", { value: jest.fn() });
+Object.defineProperty(zowe.Upload, "pathToDataSet", { value: jest.fn() });
+Object.defineProperty(vscode.window, "showErrorMessage", { value: jest.fn() });
+Object.defineProperty(vscode.window, "showInformationMessage", { value: jest.fn() });
+Object.defineProperty(vscode.window, "showWarningMessage", { value: jest.fn() });
+Object.defineProperty(vscode.window, "showInputBox", { value: jest.fn() });
+Object.defineProperty(vscode.workspace, "openTextDocument", { value: jest.fn() });
+Object.defineProperty(vscode.window, "showTextDocument", { value: jest.fn() });
+Object.defineProperty(vscode.window, "showQuickPick", { value: jest.fn() });
+Object.defineProperty(vscode.commands, "executeCommand", { value: jest.fn() });
+Object.defineProperty(globals, "LOG", { value: jest.fn() });
+Object.defineProperty(globals.LOG, "debug", { value: jest.fn() });
+Object.defineProperty(globals.LOG, "error", { value: jest.fn() });
+Object.defineProperty(zowe, "Download", { value: jest.fn() });
+Object.defineProperty(zowe.Download, "dataSet", { value: jest.fn() });
+Object.defineProperty(zowe, "Delete", { value: jest.fn() });
+Object.defineProperty(zowe.Delete, "dataSet", { value: jest.fn() });
+Object.defineProperty(fs, "unlinkSync", { value: jest.fn() });
+Object.defineProperty(fs, "existsSync", { value: jest.fn() });
+Object.defineProperty(sharedUtils, "concatChildNodes", { value: jest.fn() });
+Object.defineProperty(Profiles, "getInstance", { value: jest.fn() });
+Object.defineProperty(zowe, "List", { value: jest.fn() });
+Object.defineProperty(zowe.List, "dataSet", { value: jest.fn() });
+Object.defineProperty(vscode, "ProgressLocation", { value: jest.fn() });
+Object.defineProperty(vscode.window, "createWebviewPanel", { value: jest.fn() });
+Object.defineProperty(vscode.env, "clipboard", { value: clipboard });
+
+// Idea is borrowed from: https://github.com/kulshekhar/ts-jest/blob/master/src/util/testing.ts
+const mocked = (fn: any): jest.Mock => fn;
+
+describe("Add createMember tests", () => {
+    function generateEnvironmentalMocks() {
+        const session = generateISession();
+        const imperativeProfile = generateIProfile();
+        const zosmfSession = generateBasicZosmfSession(imperativeProfile);
+        const treeView = generateTreeView();
+        const datasetSessionNode = generateDatasetSessionNode(session, imperativeProfile);
+        const mvsApi = generateMvsApi(imperativeProfile);
+        bindMvsApi(mvsApi);
+
+        return {
+            session,
+            zosmfSession,
+            treeView,
+            imperativeProfile,
+            datasetSessionNode,
+            mvsApi,
+            testDatasetTree: generateDatasetTree(datasetSessionNode, treeView)
+        };
+    }
```

Comment:
Can we have a one main describe() block that defines this `function generateEnvironmentalMocks()` rather than add it in all describe blocks?

Reply:
No need for a describe block, just move the function definition to a module scope (unless there is a difference among the the individual implementations of `generateEnvironmentalMocks` - I do not see any).